lang/src/main/java/com/sap/psr/vulas/backend/BackendConnector.java

```diff
@@ -470,6 +468,25 @@ public boolean uploadTouchPoints(GoalContext _goal_context, Application _app, St
 		}
 		return constructs;
 	}
+	
+	/**
+	 * Returns all {@link Dependency}s of the given {@link Application} including their reachable {@link ConstructId}s.
+	 */
+	public Set<Dependency> getAppDependencies(GoalContext _ctx, @NotNull Application _app) throws BackendConnectionException {
```

Comment:
couldn't we reuse the existing method "getAppDeps" ?

Reply:
The JsonView currently used by that endpoint does not contain the reachable constructs of the dependencies. We would need to select the JsonView dynamically, probably using a new request parameter and then proceed as described here: https://stackoverflow.com/questions/28878488/dynamic-selection-of-jsonview-in-spring-mvc-controller/29103449#29103449

Not sure that this is a better solution...